### PR TITLE
Revise typedefs to support integration

### PR DESCRIPTION
**Pull request by DanielHeath**
I've added these typedefs to get a fair-sized project switched over. Perhaps they're useful to you too?

### CODE REVIEW

File: compat/src/index.d.ts [hunk 1]
```diff
@@ -21,7 +21,7 @@ declare namespace React {
 	export import StateUpdater = _hooks.StateUpdater;
 	export import useCallback = _hooks.useCallback;
 	export import useContext = _hooks.useContext;
-	export import useDebugValue = _hooks.useDebugValue;
```
Comment (JoviDeCroock) [inline compat/src/index.d.ts:24]: We probably don't wanna remove this

Comment (DanielHeath) [reply compat/src/index.d.ts:24]: Ooh that's not meant to be deleted. For that matter, I don't expect that this PR is necessarily "production grade"; it merely meets my needs.

**Comment by developit** [comment]
This actually looks like its worth landing pretty much as-is, minus the `useDebugValue` removal.

**Comment by JoviDeCroock** [comment]
Closing due to #3534

**Pull request closed without merging**